### ISSUE TITLE
[public-1.10.0] Crash: server.dll + 0x1215d0

### ISSUE THREAD

**Issue by dysphie** [nmrih/source-game#893]
[Dumps (ZIP)](https://github.com/nmrih/source-game/files/1827649/Minidumps.zip)

What we know:

- Affects maps with constant I/O the most. Things like using logic timers to make a light flicker twice a second will make the map a lot more crash-prone ..I think?
- Happens mostly on map reset or change, but not exclusively.
- Affects only NMO.
- Average recurrence is once a day.

Tested builds:
- 1.09.6 [svn-r8572] for Windows (at server.dll@0x1c47b1)
- 1.10.0 [svn-r8886] for Windows (at server.dll@0x1215d0)
- v1.10.1-r4784 [svn-r8913] for Windows (at server.dll@0x130560)

**Comment by DDysphie** [comment]
Possibly related: [hl2_1100_crash_2018_4_29T21_1_42C0.zip](https://github.com/nmrih/source-game/files/1980882/hl2_1100_crash_2018_4_29T21_1_42C0.zip)

Crash on dev while playing NMO on a listen server, pre-round phase, just as the counter reached 0:00. Time of occurrence seems typical of 0x1c47b1.

**Comment by Bubka3** [comment]
Retest after src-4571 required

**Comment by Dysphie** [comment]
Still happens as server.dll + 0x1215d0

**Comment by Dysphie** [comment]
Since it was discovered that this happens after the event queue being emptied prematurely, could we add some developer 2 verbose to its clear method? I wanna be able to tell if it's actually `Clear()` being called; seems unlikely..

**Comment by Pizhik** [comment]
Also server.dll@0x1215d0

https://crash.limetech.org/gbiydcexhygr
https://crash.limetech.org/pghbq5yeac3h
https://crash.limetech.org/6dtdga34a3a5
https://crash.limetech.org/qdvfsxh4himo
https://crash.limetech.org/334sb2klem2g
https://crash.limetech.org/n2z3rbej5bm3
https://crash.limetech.org/k5d47m6zydui
https://crash.limetech.org/lmbv2wmq5tam
https://crash.limetech.org/evtgycitahi7
https://crash.limetech.org/ntoyn4gtijgu
https://crash.limetech.org/jgi2m4bx3hzu
https://crash.limetech.org/lnfbf3lb3yvd
https://crash.limetech.org/jminxco5e5ys
https://crash.limetech.org/pwzssgessdr7
https://crash.limetech.org/ych2kzw7s33d
https://crash.limetech.org/7eeybj7ykeob
https://crash.limetech.org/ehzqpwpvki2b
https://crash.limetech.org/oaw5mczp567g
https://crash.limetech.org/gaxwitgbpfjb
https://crash.limetech.org/h5t6itbcnish
https://crash.limetech.org/43pz2lxf4ixo

**Comment by Pizhik** [comment]
Ok, I fucked up.

@Bubka3 
Dumps:
https://github.com/Pizhik/pizhiks_nmrih_mods/blob/master/0x1215d0.zip

**Comment by Pizhik** [comment]
Also affect Linux:
https://github.com/Pizhik/pizhiks_nmrih_mods/blob/master/dumps_linux.zip

SourceMod 1.9.0.6260, by AlliedModders LLC
Metamod:Source version 1.10.7-dev
Compiled on: Dec 19 2018
Plugin interface version: 15:14
SourceHook version: 5:5
Steam Client Version 1100
Steam Patch Version 1.1.0.0
Steam Product Name nmrih
Steam Server App ID 317670
Steam Server Version 1100
Steam App ID 224260

**Comment by dysphie** [comment]
Okay here's a rather consistent way to reproduce this:

1. Start a game of nmo_resident_evil_revelationv2_v4
2. Use these to save time:

```
callvote restart
sv_preround_freezetime 0
sv_postround_time 0
sv_gravity 50000
```


3. Noclip towards the ventilation room, while staying in-bounds, and just drop dead repeatedly.

The game should crash after ~10 restarts.
Video: https://streamable.com/s8ddu


Now for the interesting bit: if you do this with AI disabled, the game will never crash.

Bonus ZPS issue, we're crashing at the same location: ValveSoftware/source-sdk-2013#388

**Comment by essbuh** [comment]
These crashes have likely all been fixed in code r4833 and should be included in 1.10.1.

**Comment by felis-catus** [comment]
Confirmed fixed in src-r4833.
